src/main/java/org/elasticsearch/indices/cache/query/IndicesQueryCache.java

```diff
@@ -218,7 +221,7 @@ public boolean canCache(ShardSearchRequest request, SearchContext context) {
      * to have a single load operation that will cause other requests with the same key to wait till its loaded an reuse
      * the same cache.
      */
-    public QuerySearchResultProvider load(final ShardSearchRequest request, final SearchContext context, final QueryPhase queryPhase) throws Exception {
+    public void load(final ShardSearchRequest request, final SearchContext context, final QueryPhase queryPhase) throws Exception {
```

Comment:
now that it doesn't return a value, maybe rename in to loadIntoContext , or add something to the java docs to indicate where the output is put.